src/main/java/com/cryptomorin/xseries/skull/Action.java

```diff
@@ -9,9 +9,20 @@
 import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.atomic.AtomicInteger;
 
+/**
+ * The {@code Action} class represents an action that can be performed on a {@link GameProfile}.
+ * It uses an {@link Instruction} to define how the action is applied.
+ *
+ * @param <T> The type of the result produced by the action.
+ */
 public class Action<T> {
```

Comment:
Is there a point in having Instruction and Action as separate classes? We're only going to be using it for XSkull.
Also just for the sake of IDE tab-completion clutter, you should prefix all these new classes with `Skull`